### DISCUSSION TITLE
Do dynamic pages automatically redirect to the not-found page if they are not created by the generateStaticParams function?

### DISCUSSION THREAD

**Discussion by GetWrecked99** [vercel/next.js#66741] [Help] [answered]
### Summary

Hi, 
this is my routing structure: 

```
   -app
      -products
         -[slug]
            page.tsx
```

Inside the dynamic page, i have this sample: 

```
export const generateStaticParams = () => ['1', '2', '3', '4', '5'].map((slug) => ({slug}))

const page = () => {
  return <div>detail page</div>
}

export default page
```

When we build the app, it generates this routes: 

```
   - /products/1
   - /products/2
   - /products/3
   - /products/4
   - /products/5
```

After running the application, we can navigate to any other routes like /products/6 & so on . . .  and it does not trigger the notFound page by default.

Just wanted to know that this behavior is not done by default?

**Comment by icyJoseph** [comment, marked as answer]
I think you need to set: https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamicparams

```
- true (default): Dynamic segments not included in generateStaticParams are generated on demand.
- false: Dynamic segments not included in generateStaticParams will return a 404.
```

**Reply by GetWrecked99** [reply]
Appreciated mate.
